python/ray/serve/drivers.py

```diff
@@ -174,3 +113,68 @@ async def get_intermediate_object_refs(self) -> Dict[str, Any]:
     async def get_dag_node_json(self) -> str:
         """Returns the json serialized root dag node"""
         return self.dags[self.MATCH_ALL_ROUTE_PREFIX].dag_node_json
+
+
+@PublicAPI(stability="alpha")
+class gRPCIngress:
+    """
+    gRPC Ingress that starts gRPC server based on the port
+    """
+
+    def __init__(self, port=DEFAULT_GRPC_PORT):
+        self.server = grpc.aio.server()
+        self.port = port
+
+        # protobuf Schema gRPC should generate bind function
+        # (e.g. add_PredictAPIsServiceServicer_to_server) to bind gRPC server
+        # and schema interface
+        bind_function_name = "add_{}_to_server"
+        module_name = self.__class__.__bases__[1].__module__
+        servicer_name = self.__class__.__bases__[1].__name__
+        getattr(sys.modules[module_name], bind_function_name.format(servicer_name))(
+            self, self.server
+        )
```

Comment:
make this into a unit testable thing? and test it. this is some sort of hacky convention that we reply on for gRPC package, and it can potentially change.

Reply:
not addressed